src/components/Identity/createIdSyncs.js

```diff
@@ -37,58 +37,68 @@ const setControlObject = (controlObject, cookieJar) => {
 };
 
 const createProcessor = (config, logger, cookieJar) => destinations => {
-  if (!config.idSyncsEnabled) {
-    return;
-  }
-
-  const controlObject = getControlObject(cookieJar);
-  const now = convertTimes(MILLISECOND, HOUR, new Date().getTime()); // hours
-
-  Object.keys(controlObject).forEach(key => {
-    if (controlObject[key] < now) {
-      delete controlObject[key];
+  return new Promise(resolve => {
+    if (!config.idSyncsEnabled) {
+      return;
     }
-  });
 
-  const idSyncs = destinations
-    .filter(dest => dest.type === "url" && controlObject[dest.id] === undefined)
-    .map(dest =>
-      assign(
-        {
-          id: dest.id
-        },
-        dest.spec
-      )
-    );
-
-  if (idSyncs.length) {
-    fireDestinations({
-      logger,
-      destinations: idSyncs
-    }).then(result => {
-      const timeStamp = Math.round(
-        convertTimes(MILLISECOND, HOUR, new Date().getTime())
-      ); // hours
-
-      result.succeeded.forEach(idSync => {
-        const ttl = (idSync.ttl || 7) * 24; // hours
-
-        if (idSync.id !== undefined) {
-          controlObject[idSync.id] = timeStamp + ttl;
-        }
-      });
+    const controlObject = getControlObject(cookieJar);
+    const now = convertTimes(MILLISECOND, HOUR, new Date().getTime()); // hours
 
-      setControlObject(controlObject, cookieJar);
+    Object.keys(controlObject).forEach(key => {
+      if (controlObject[key] < now) {
+        delete controlObject[key];
+      }
+    });
 
-      cookieJar.set(
-        ID_SYNC_TIMESTAMP,
-        (
-          Math.round(convertTimes(MILLISECOND, HOUR, new Date().getTime())) +
-          convertTimes(DAY, HOUR, 7)
-        ).toString(36)
+    const idSyncs = destinations
+      .filter(
+        dest => dest.type === "url" && controlObject[dest.id] === undefined
+      )
+      .map(dest =>
+        assign(
+          {
+            id: dest.id
+          },
+          dest.spec
+        )
       );
-    });
-  }
+
+    if (idSyncs.length) {
+      fireDestinations({
+        logger,
+        destinations: idSyncs
+      }).then(result => {
+        const timeStamp = Math.round(
+          convertTimes(MILLISECOND, HOUR, new Date().getTime())
+        ); // hours
+
+        result.succeeded.forEach(idSync => {
+          const ttl = Math.round(
+            convertTimes(MINUTE, HOUR, idSync.ttlMinutes || 10080)
+          ); // hours
+
+          if (idSync.id !== undefined) {
```

Comment:
I don't know that you need this check; you already filtered the syncs by `id` here: https://github.com/adobe/alloy/pull/171/commits/3f5f04b0cb41822f62fb11237e33d2afaf8691e5#diff-282e825508ed24a8608c10a349958136R56